umh-core/pkg/fsm/agent/actions.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agent
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/metrics"
+	agent_service "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/agent"
+)
+
+// The functions in this file define heavier, potentially fail-prone operations
+// that the Agent FSM might need to perform. They are intended to be called from Reconcile.
+//
+// Each action is expected to be idempotent, since it may be retried multiple times.
+// Each action takes a context.Context and can return an error if the operation fails.
+
+// initiateAgentActivation attempts to activate the agent monitoring.
+func (a *AgentInstance) initiateAgentActivation(ctx context.Context) error {
+	a.baseFSMInstance.GetLogger().Debugf("Starting Action: Activating agent monitoring...")
+
+	// Initialize the agent with current configuration
+	err := a.service.Initialize(ctx, a.config)
+	if err != nil {
+		return fmt.Errorf("failed to initialize agent: %w", err)
+	}
+
+	a.baseFSMInstance.GetLogger().Debugf("Agent monitoring activated successfully")
+	return nil
+}
+
+// initiateAgentMonitoringStopped attempts to stop agent monitoring.
+// In a real implementation, this would include cleanup and stopping resources.
+func (a *AgentInstance) initiateAgentMonitoringStopped(ctx context.Context) error {
+	a.baseFSMInstance.GetLogger().Debugf("Starting Action: Stopping agent monitoring...")
+
+	// In a real implementation, you would call service methods to stop resources
+	// For now, we'll just log it
+	a.baseFSMInstance.GetLogger().Debugf("Agent monitoring stopped successfully")
+	return nil
+}
+
+// checkAgentHealth verifies if the agent is healthy or degraded
+func (a *AgentInstance) checkAgentHealth(ctx context.Context) (healthy bool, err error) {
+	a.baseFSMInstance.GetLogger().Debugf("Checking agent health...")
+
+	// In a real implementation, we would perform actual health checks
+	// For now, we'll simulate based on the observed state
+	if a.ObservedState.AgentInfo.Status == agent_service.AgentStatusError {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// checkAndUpdateConfig checks for updated configuration and applies any changes
```

Comment:
an instance configuration is automatically updated through its manager, so this function is not needed here